ext/tidy/tidy.c

```diff
@@ -221,6 +208,19 @@ static void php_tidy_load_config(TidyDoc doc, const char *path)
 	}
 }
 
+static zend_result php_tidy_apply_config(TidyDoc doc, zend_string *str_string, HashTable *ht_options)
+{
+	if (ht_options) {
+		return _php_tidy_apply_config_array(doc, ht_options);
+	} else if (str_string) {
+		if (php_check_open_basedir(ZSTR_VAL(str_string))) {
+			return FAILURE;
+		}
+		php_tidy_load_config(doc, ZSTR_VAL(str_string));
+	}
```

Comment:
Note to myself, see if the HashTable and underlying char pointer can be made const.